Day3Part2.js

```diff
@@ -0,0 +1,53 @@
+const readline = require("readline").createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+readline.question(`enter your data `, function LifeSupportRating(data) {
+  var Data = data.split(" ");
+  zeroCounter = 0;
+  oneCounter = 0;
+
+  let arro2 = [...Data];
+  let arrco2 = [...Data];
+
+  let i = 0;
+  
+  const getMostCommun = (array, i) => {
+    for (let j = 0; j < array.length; j++) {
+      if (Data[j][i] === "0") {
+        zeroCounter += 1;
+      } else {
+        oneCounter += 1;
+      }
+    }
+    if (oneCounter > zeroCounter || oneCounter == zeroCounter) {
+      return 1;
+    } else {
+      return 0;
+    }
+    zeroCounter = 0;
+    oneCounter = 0;
+  };

```

Comment:
The lines after the if condition will not be executed because there is a `return statement` this is why the second time you run the function zeroCounter and oneCounter still have their values. If you don't understand why you should read more about ["scopes" and "closures"](https://javascript.info/closure).

```suggestion
  const getMostCommun = (array, i) => {
    let zeroCounter = 0
    let oneCounter = 0
 
    for (let j = 0; j < array.length; j++) {
      if (Data[j][i] === "0") {
        zeroCounter += 1;
      } else {
        oneCounter += 1;
      }
    }
    if (oneCounter > zeroCounter || oneCounter == zeroCounter) {
      return 1;
    } else {
      return 0;
    }
  };

```